tinywl/tinywl.c

```diff
@@ -762,6 +777,8 @@ static void xdg_toplevel_destroy(struct wl_listener *listener, void *data) {
 	wl_list_remove(&toplevel->request_maximize.link);
 	wl_list_remove(&toplevel->request_fullscreen.link);
 
+	wlr_scene_node_destroy(&toplevel->border->node);
+	wlr_scene_node_destroy(&toplevel->shadow->node);
```

Comment:
Isn't needed. Automatically called when its parent is destroyed, the line below :)